_posts/2020-01-21-hhvm-4.41.markdown

```diff
@@ -0,0 +1,40 @@
+---
+title: "HHVM 4.41"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.41 is released! This release marks the end of support for 4.34;
+HHVM 4.35&ndash;4.40 remain supported, as do the 4.8 and 4.32 LTS releases.
+
+# Highlights
+
+- A new type alias, `HH\enumname<TEnum>`, represents the name of the enum type
+  `TEnum` (similarly to `classname<TClass>`).
+- A new function, `HH\type_structure_for_alias`, returns the `TypeStructure` of
+  the specified type alias (similarly to `HH\type_structure` for type constants
+  in classes).
+  - Note that the `TypeStructure` API remains experimental and may be changed or
+    removed in the future.
+- Builtin functions `usort`, `uasort`, `uksort` now accept callbacks that return
+  any `num` value (`int` or `float`), allowing for more convenient sorting of
+  `float` arrays.
+
+# Breaking Changes
+
+- Fixed a bug that allowed XHP children declarations to contain undefined types.
+  This may reveal many previously ignored errors.
+  - XHP children declarations can now only contain defined, non-aliased,
+    non-generic Hack types, XHP category names (`%foo`), and the special values
+    `pcdata`, `any` and `empty`.
```

Comment:
probably also worth highlighting that these special values are case-sensitive, as are type names

Reply:
I thought it was well known that everything is case-sensitive (in the typechecker at least), but you're right I guess it's useful to mention here since people might see errors with PCData etc.

Reply:
These were previously unchecked, and aren't actual type names; we had quite a few cases of `PCDATA` in particular